### PR TITLE
bug fix to Inconsistent comment for EXIF issue id #15706

### LINKED ISSUE #15706: Inconsistent comment for EXIF
##### System information (version)

- OpenCV => master

##### Detailed description

This line of comment is inconsistent with the code:
https://github.com/opencv/opencv/blob/174b4ce29d8e1ddbd899095c4b9fb4443444af45/modules/imgcodecs/src/exif.hpp#L157

Constructor has been changed to use istream instead of filename in this PR:
https://github.com/opencv/opencv/pull/8492

### PR DESCRIPTION
**Pull request by jshiwam**
resolves #15706

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/imgcodecs/src/exif.hpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ enum ImageOrientation
  * Usage example for getting the orientation of the image:
  *
  *      @code
- *      ExifReader reader(fileName);
+ *      ExifReader reader(std::cin);
```
Comment (paroj) [inline modules/imgcodecs/src/exif.hpp]: this should rather be an std::ifstream

Comment (jshiwam) [reply modules/imgcodecs/src/exif.hpp]: Okay

File: modules/imgcodecs/src/exif.hpp [hunk 2]
```diff
@@ -154,7 +154,7 @@ enum ImageOrientation
  * Usage example for getting the orientation of the image:
  *
  *      @code
- *      ExifReader reader(fileName);
+ *      ExifReader reader(std::ifstream);
```
Comment (alalek) [inline modules/imgcodecs/src/exif.hpp]: This code can not compile.

Comment (jshiwam) [reply modules/imgcodecs/src/exif.hpp]: Can you please mention the error, the code builds on my laptop

Comment (alalek) [reply modules/imgcodecs/src/exif.hpp]: Code of this snippet or OpenCV code with this comment?

Right answer should look like:
```
ExifReader reader(std::ifstream(fileName, std::ifstream::in | std::ifstream::binary));
```

Comment (jshiwam) [reply modules/imgcodecs/src/exif.hpp]: Yes this is the code that i used to test this error:
Code
#include<iostream>
#include<fstream>
#include<exif.cpp>
#include<exif.hpp>
#include<precomp.hpp>

using namespace cv;

int main(){
    std::ifstream f("/home/shiwam/Documents/opencv_test_images/shiva.jpg",std::ios_base::in | std::ios_base::binary);
    ExifReader reader(f);
    if(reader.parse()){
        int orientation = reader.getTag(ORIENTATION).field_u16;
        std::cout<<orientation;
    }else{
        std::cout<<"Failed";
    }

    

    return 0;
}

output:
1

Should I update the pr?

**Comment by alalek** [comment]
This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide]
(https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
  In this case (due merge commit) you should **squash** commits into one before rebase.
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

**Comment by alalek** [comment]
You should check commit message and its content.
You may use "gitk" (builtin git tool) locally for that.

**Comment by jshiwam** [comment]
The error produced by build bot is:

/persistence_yml.cpp:36:31: error: 'd' was not declared in this scope
             while( cv_isprint(d) && d != ' ' && d != '>' );
                               ^
/build/precommit_linux64/3.4/opencv/modules/core/src/persistence.hpp:216:36: note: in definition of macro 'cv_isprint'
 #define cv_isprint(c)     ((uchar)(c) >= (uchar)' ')
                                    ^
/build/precommit_linux64/3.4/opencv/modules/core/src/persistence_yml.cpp:42:21: error: 'value_type' was not declared in this scope
                     value_type |= CV_NODE_USER;
                     ^
/build/precommit_linux64/3.4/opencv/modules/core/src/persistence_yml.cpp: At global scope:

I checked the file persistence_yml.cpp but where I found that 'd' was declared as
do d = *++typeEndPtr;

whereas value_type is not declared.
I didn't even make changes in these files how/why do these files generate error, is it because I rebased my branch where I merged changes from others as well to resolve the conflicts.

**Comment by alalek** [comment]
> I didn't even make changes in these files

You should check contents of this patch:
- Files changed 10

You bring unnecessary/wrong code changes from other branch.

**Comment by jshiwam** [comment]
> > I didn't even make changes in these files
> 
> You should check contents of this patch:
> 
>     * Files changed 10
> 
> 
> You bring unnecessary/wrong code changes from other branch.

I Will fix that

**Comment by alalek** [comment]
> Commits 4
> Files changed 4

Please remove **unnecessary** changes from the patch. We should have 1 commit and 1 changed file here.
Commits squashing and/or commit amending may help to do that.

**Comment by jshiwam** [comment]
> > Commits 4
> > Files changed 4
> 
> Please remove **unnecessary** changes from the patch. We should have 1 commit and 1 changed file here.
> Commits squashing and/or commit amending may help to do that.

Done

**Pull request merged**